CoreUtilities/src/au/gov/asd/tac/constellation/utilities/file/FilenameEncoder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package au.gov.asd.tac.constellation.utilities.file;
+
+/**
+ * Manage the encode/decode of strings into filenames. This essentially ensures
+ * the characters in the filename are within a valid set.
+ * @author serpens24
+ */
+public class FilenameEncoder {
+
+    /**
+     * Private constructor to hide implicit public one.
+     */
+    private FilenameEncoder() {
+        throw new IllegalStateException("Static class");
```

Comment:
I might need some clarification here. You mention in the exception that it is a static class but the class itself doesn't appear to be static

Reply:
good point. This change was made based on a sonarqube code smell, basically to prevent a default public constructor getting called. I'll fix